packages/sdk/docs/Developers/Creating-Policies.md

```diff
@@ -0,0 +1,645 @@
+        maxDailySpendingLimit: dailySpendingLimit,
+        currentDailySpending,
+        allowedTokens: allowedTokens,
+      });
+    }
+
+    return policyContext.allow({
+      maxDailySpendingLimit: dailySpendingLimit,
+      currentDailySpending,
+      allowedTokens: allowedTokens,
+    });
+  },
+});
+```
+
+Two arguments are passed to your policy's `evaluate` function by the Vincent Tool & Policy SDK. The first is an object containing the `toolParams` and `userParams` the adhere to the `toolParamsSchema` and `userParamsSchema` you have defined for your policy. The second is the [`policyContext`](#the-policycontext-argument) managed by the Vincent Tool & Policy SDK that contains helper methods for returning `allow` and `deny` results, as well as some metadata about the Vincent App that the policy is being executed for.
+
+### `evalAllowResultSchema`
+
+This Zod schema defines the structure of a successful `evaluate` result. What's included in the returned object is up to you, but ideally it includes details about why the `evaluate` function is allowing the Vincent Tool execution.
+
+The following schema returns useful information to the Vincent Tool executor about the current policy state for the Vincent App User and Vincent App combo:
+
+```typescript
+import { createVincentPolicy } from '@lit-protocol/vincent-tool-sdk';
+import { z } from 'zod';
+
+const vincentPolicy = createVincentPolicy({
+  // ... other policy definitions
+
+  evalAllowResultSchema: z.object({
+    maxDailySpendingLimit: z.number(),
+    currentDailySpending: z.number(),
+    allowedTokens: z.array(z.string()),
+  }),
+});
+```
+
+### `evalDenyResultSchema`
+
+This Zod schema defines the structure of a denied `evaluate` result. What's included in the returned object is up to you, but ideally it includes details about why the `evaluate` function is denying the Vincent Tool execution.
+
+The following schema returns additional information to the Vincent Tool executor that would allow the Vincent Tool executor to adapt their execution request so that the policy's `evaluate` validation checks don't fail:
+
+```typescript
+import { createVincentPolicy } from '@lit-protocol/vincent-tool-sdk';
+import { z } from 'zod';
+
+const vincentPolicy = createVincentPolicy({
+  // ... other policy definitions
+
+  evalDenyResultSchema: z.object({
+    reason: z.string(),
+    maxDailySpendingLimit: z.number(),
+    currentDailySpending: z.number(),
+    allowedTokens: z.array(z.string()),
+  }),
+});
+```
+
+## Commit Function
+
+The `commit` function is called by the Vincent Tool's `execute` method after all Vincent Policies registered by the Vincent App User for the Vincent Tool have been evaluated and returned `allow` results, and the Vincent Tool's execution logic has been successfully executed. This functions is **optional**, and is only required if you need to update any state that the policy depends on, after the Vincent Tool executes.
```

Comment:
> The `commit` function is called by the Vincent Tool's `execute` method after all Vincent Policies registered by the Vincent App User for the Vincent Tool have been evaluated and returned `allow` results, and the Vincent Tool's execution logic has been successfully executed.

While this is the intended order of when the `commit` functions are executed, I don't think this is enforced and the Tool LA can call the commit methods whenever, or completely ignore them